common/v2/libs/networks/networks.ts

```diff
@@ -1,19 +1,40 @@
-import { NetworkSelect } from './types';
 import { getCache } from 'v2/services/LocalCache';
 import { NetworkOptions } from 'v2/services/NetworkOptions/types';
 
-export const getNetworkByChainId = (chainId: string): NetworkSelect => {
-  const networks = getAllNetworks();
+export const getAllNetworks = () => {
+  return Object.values(getCache().networkOptions);
+};
 
-  let networkToSelect = null;
-  networks.map((network: NetworkOptions) => {
-    if (network.chainId === parseInt(chainId, 16)) {
-      networkToSelect = network;
+export const getNetworkByChainId = (chainId: string): Promise<NetworkOptions> => {
+  return new Promise((resolve, reject) => {
+    try {
+      const networks = getAllNetworks();
+
+      networks.map((network: NetworkOptions) => {
+        if (network.chainId === parseInt(chainId, 16)) {
+          resolve(network);
+        }
+      });
+      reject();
+    } catch (e) {
+      reject();
     }
   });
-  return networkToSelect === null ? null : networkToSelect;
 };
 
-export const getAllNetworks = () => {
-  return Object.values(getCache().networkOptions);
+export const getNetworkByName = async (name: string): Promise<NetworkOptions> => {
+  return new Promise((resolve, reject) => {
+    try {
+      const networks = getAllNetworks();
+
+      networks.map((network: NetworkOptions) => {
+        if (network.name === name) {
+          resolve(network);
+        }
+      });
+      reject();
+    } catch (e) {
+      reject();
+    }
+  });
 };
```

Comment:
The Promise constructor's resolve function can only be called once, but it's called in map, which would get called for each network in networks. What is your intention here? Do you want to resolve the promise with an array of all networks, or just get the first one? Either way, you shouldn't need to use the Promise constructor for syncronous code.